lib/DXIL/DxilShaderModel.cpp

```diff
@@ -504,8 +504,9 @@ bool ShaderModel::AllowDerivatives(DXIL::ShaderKind sk) const {
   case DXIL::ShaderKind::Amplification:
   case DXIL::ShaderKind::Mesh:
     return IsSM66Plus();
+  default:
+    return false;
   }
-  return false;
```

Comment:
If this is unreachable we should have an `llvm_unreachable` here, otherwise as code changes if a case breaks out instead of returning this becomes UB.

Reply:
How about I just change this to handle all cases (no default) and with the `llvm_unreachable` to be sure we added all our cases here?

Also, we don't appear to be building with this warning enabled, but I thought we hit this before in some other scenario.  Perhaps we aren't warning clean across the project with this one, but it would be good to be warning clean for the DXIL headers/cpp files that are consumed by other projects at a minimum.